ray-curator/tests/stages/test_base.py

```diff
@@ -0,0 +1,102 @@
+"""Tests for base stage classes."""
+
+from dataclasses import dataclass
+
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.resources import Resources
+from ray_curator.tasks import Task
+
+
+class MockTask(Task[dict]):
+    """Mock task for testing."""
+
+    def __init__(self, data: dict | None = None):
+        super().__init__()
+        self.data = data or {}
+
+    @property
+    def num_items(self) -> int:
+        return len(self.data)
+
+    def validate(self) -> bool:
+        return True
+
+
+@dataclass
+class ConcreteProcessingStage(ProcessingStage[MockTask, MockTask]):
```

Comment:
Your Concrete  in test case is a `dataclass` (this is related class variable vs instance, when you decorate with @dataclass those are overridden as instance variables, vs when you remove @dataclass it still works, because you have overridden the class variable)